src/types/message_entity.rs

```diff
@@ -39,21 +39,90 @@ pub enum MessageEntityKind {
     Strikethrough,
 }
 
-#[test]
-fn recursive_kind() {
-    use serde_json::from_str;
+impl MessageEntity {
+    pub fn text_from(&self, message: Message) -> Option<String> {
+        let text= message.text();
+        match text {
```

Comment:
You could use `?` instead of `match`: https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=683e507a9dab24898d5fc6077a290e55

Reply:
fixed